client/src/main/java/org/apache/celeborn/client/read/ChunkClient.java

```diff
@@ -256,7 +252,7 @@ public void onFailure(int chunkIndex, Throwable e) {
         initiateRetry(chunkIndex, this.currentNumTries);
       } else {
         logger.error("Abandon to fetch chunk {} after {} tries.", chunkIndex, this.currentNumTries);
-        callback.onFailure(chunkIndex, e);
+        callback.onFailure(chunkIndex, new ChunkClientException(e.getMessage(), e, location));
```

Comment:
We should make location a parameter or onFailure instead of ChunkClientException